### PR TITLE
back/frontend unit converter

### PR DESCRIPTION
**Pull request by Mathieu-Deharbe**
This PR is a more generic version of this one : https://github.com/gridsuite/gridstudy-app/pull/1876 Only one of those should be merged.

It adds a js file specific to  unit conversion and uses it in the security analysis and shortcircuit result aggrid tables.

The point is to have the same values in the front and back but to display them differently if the dev adds a single `displayConversionMode` definition line in the custom aggrid header.

### CODE REVIEW

File: src/utils/unit-converter.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { kiloUnitToUnit, unitToKiloUnit } from './rounding';
```
Comment (dbraquart) [inline src/utils/unit-converter.tsx:1]: missing header/copyright

Comment (Mathieu-Deharbe) [reply src/utils/unit-converter.tsx]: Done. I also moved the unit conversion functions to the more specific unit-converter.jsx file.

File: src/utils/unit-converter.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/**
```
Comment (lilian-houdelet) [inline src/utils/unit-converter.tsx:1]: No need to do a tsx file, tsx is for react gestion, .ts should work

File: src/components/dialogs/network-modifications/line/modification/line-modification-dialog.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ import {
 } from 'components/utils/field-constants';
 import { FormProvider, useForm } from 'react-hook-form';
 import { sanitizeString } from 'components/dialogs/dialogUtils';
-import { microUnitToUnit, unitToMicroUnit } from 'utils/rounding';
+import { microUnitToUnit, unitToMicroUnit } from 'utils/unit-converter.tsx';
```
Comment (lilian-houdelet) [inline src/components/dialogs/network-modifications/line/modification/line-modification-dialog.js:31]: why .tsx in import ? all the others are 'utils/unit-converter'

**Review by lilian-houdelet — CHANGES_REQUESTED** [review]

File: src/utils/unit-converter.tsx (hunk 2 above)
Comment (Mathieu-Deharbe) [reply src/utils/unit-converter.tsx]: Indeed, renamed.

File: src/components/dialogs/network-modifications/line/modification/line-modification-dialog.js (hunk 3 above)
Comment (Mathieu-Deharbe) [reply src/components/dialogs/network-modifications/line/modification/line-modification-dialog.js]: Useless indeed. `.tsx` removed from 3 files.

File: src/module-ag-grid-community.d.ts [hunk 4]
```diff
@@ -6,12 +6,14 @@
  */
 
 import { CrossValidationOptions } from './components/spreadsheet/utils/equipment-table-utils';
+import {DISPLAY_CONVERSION} from "./components/custom-aggrid/custom-aggrid-header-utils";
```
Comment (lilian-houdelet) [inline src/module-ag-grid-community.d.ts:9]: the syntax is not the same than the line above

Comment (Mathieu-Deharbe) [reply src/module-ag-grid-community.d.ts]: Ok done.

**Review by dbraquart — APPROVED** [review]
code: ok
tests: ok (short-circuit display and filtering on IccMin/Max)

**Review by lilian-houdelet — APPROVED** [review]

**Pull request merged**